docs/src/explanations/Documentation-of-functions.md

```diff
@@ -0,0 +1,404 @@
+  // connect Producer to IO
+  p.io.a <> io.in
+  // connect producer to consumer
+  c.io.a <> p.io.b
+  // connect consumer to IO
+  io.out <> c.io.b
+}
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.b <> io.a
+}
+```
+Below we can see the resulting Verilog for this example:
+```scala modc
+ChiselStage.emitVerilog(new Wrapper)
+```
+## Concept 1: <> is Commutative
+
+
+
+This experiment is set up to test for the function of `<>` using the experiment above.
+
+Achieving this involves flipping the RHS and LHS of each arrow and seeing how '<>'  will react.
+( Scastie link for the experiment:https://scastie.scala-lang.org/Shorla/LVhlbkFQQnq7X3trHfgZZQ )
+
+
+
+
+```scala mdoc:silent:reset
+import chisel3._
+import chisel3.util.DecoupledIO
+
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(DecoupledIO(UInt(8.W)))
+  val out = DecoupledIO(UInt(8.W))
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  // connect Producer to IO
+  io.in := p.io.a
+  // connect producer to consumer
+  p.io.b := c.io.a
+  // connect consumer to IO
+  c.io.b := io.out
+}
+
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.a <> io.b
+}
+```
+Below we can see the resulting Verilog for this example:
+```scala modc
+ChiselStage.emitVerilog(new Wrapper)
+```
+### Conclusion: 
+The Verilog remained the same without incurring errors, showing that the `<>` operator is generally commutative.
+
+
+
+
+## Concept 2: := means assign ALL signals from the RHS to the LHS, regardless of their direction.
+Using the same experiment code as above, we set to test for the function of ":="
+To achieve this, "<>" is being replaced with ":=" in the sample code above.
+(Scastie link to the experiment:https://scastie.scala-lang.org/Shorla/o1ShdaY3RWKf0IIFwwQ1UQ)
+
+```scala mdoc:silent:reset
+import chisel3._
+import chisel3.util.DecoupledIO
+
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(DecoupledIO(UInt(8.W)))
+  val out = DecoupledIO(UInt(8.W))
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  // connect Producer to IO
+  p.io.a <> io.in
+  // connect producer to consumer
+  c.io.b <> p.io.b
+  // connect consumer to IO
+  io.out <> c.io.b
+}
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.b := io.a
+}
+```
+Below we can see the resulting error message for this example:
+```scala modc:crash
+ChiselStage.emitVerilog(new Wrapper)
+```
+### Conclusion:
+The `:=` operator goes field-by-field and attempts to connect the RHS to the LHS. If something on the LHS is actually an `Input`, or something on the RHS is an `Output`, you will get this error:
+
```

Comment:
we are no longer showing the error here, what is the error we get?

Reply:
This is the error we got:
chisel3.internal.ChiselException: Connection between sink (DecoupledIO(IO io_a in PipelineStage)) and source (DecoupledIO(IO in unelaborated Wrapper)) failed @.ready: Sink is unwriteable by current module.
	at ... ()
	at Playground$Wrapper.<init>(main.scala:15)
	at Playground$$anonfun$13.apply(main.scala:31)
	at Playground$$anonfun$13.apply(main.scala:31)
	at ... ()
	at ... (Stack trace trimmed to user code only. Rerun with --full-stacktrace to see the full stack trace)

Reply:
I removed the errors because I did the same for other examples with errors.
Should I add the error or Just rewrite the conclusion, so it doesn't require adding the error message?

Reply:
I think you can include the mdoc-generated error message below, or say something like "the error message shown above"... as it was written it seemed like the error would be shown below.